standard/flags.lua

```diff
@@ -62,6 +62,50 @@ function Flags.Icon(args, flagName)
 	end
 end
 
+-- Returns the localisation/country-adjective of a country or region
+--[[
+supported args are:
+country		- country name, flag code, or alias of the Flag
+hideError	- boolean that decides if there should be a displayed error if no entry is found
+simpleError	- boolean that decides if displayed error should be simple or detailed
+examples:
+Flags.getLocalisation({country = de}) = German
+Flags.getLocalisation({country = Germany}) = German
+Flags.getLocalisation({country = deu}) = German
+]]--
+function Flags.getLocalisation(args)
+	args = args or {}
+	local country = args.country
+
+	if String.isEmpty(country) then
+		return ''
+	end
+
+	local hideError = Logic.readBool(args.hideError)
+	local simpleError = Logic.readBool(args.simpleError)
+
+	-- clean the entered value
+	local countryKey = Flags._convertToKey(country)
+
+	if countryKey then
+		local data = MasterData.data[countryKey]
+		if String.isNotEmpty(data.localised) then
+			return data.localised
+		end
+	end
+
+	-- Return message if none is found
+	mw.log('Unknown localisation entry: ', country)
+	if hideError then
+		return ''
+	elseif simpleError then
+		return 'error'
+	else
+		return 'Unknown localisation entry "[[lpcommons:Module:Flags/MasterData' ..
+			'|' .. country .. ']][[Category:Pages with unknown countries]]'
+	end
+end
```

Comment:
```suggestion
function Flags.getLocalisation(args)
	args = args or {}
	local country = args.country

	if String.isEmpty(country) then
		return nil, 'Missing country input'
	end

	-- clean the entered value
	local countryKey = Flags._convertToKey(country)

	if countryKey then
		local data = MasterData.data[countryKey]
		if String.isNotEmpty(data.localised) then
			return data.localised
		end
	end

	-- Return message if none is found
	mw.log('Unknown localisation entry: ', country)
	return nil, 'Unknown localisation entry "[[lpcommons:Module:Flags/MasterData' ..
			'|' .. country .. ']][[Category:Pages with unknown countries]]'
end
```

Reply:
For example

Reply:
hmmm
do we actually want a warning/error return if no input was made?
i would think for that case a simple nil return would be sufficient
the rest looks good to me

will look into it tomorrow (when on pc)

Reply:
there is another major issue with your proposal:
when called from wiki code you do not get any return in case of invalid input

Reply:
how about
```lua
function Flags.getLocalisation(args)
	args = args or {}
	local country = args.country

	if String.isEmpty(country) then
		return
	end

	-- clean the entered value
	local countryKey = Flags._convertToKey(country)

	if countryKey then
		local data = MasterData.data[countryKey]
		if String.isNotEmpty(data.localised) then
			return data.localised
		end
	end

	-- Return message if none is found
	mw.log('Unknown localisation entry: ', country)
	return nil, 'Unknown localisation entry "[[lpcommons:Module:Flags/MasterData' ..
			'|' .. country .. ']]"[[Category:Pages with unknown countries]]'
end

function Flags.localisationTemplate(args)
	args = args or {}
	local display, error = Flags.getLocalisation(args)

	if not error or Logic.readBool(args.hideError) then
		return display
	end

	if Logic.readBool(args.simpleError) then
		return 'error'
	end

	return 'Unknown localisation entry "[[lpcommons:Module:Flags/MasterData' ..
		'|' .. country .. ']][[Category:Pages with unknown countries]]'
end
```

Reply:
Do we call it from Template today?

Reply:
Are there any current template calls to it?

Reply:
ehm ... new function ... so no
but to the old module possibly

Reply:
Sure your suggestion works. Let's support a template entry point